newIDE/app/src/EventsSheet/ParameterFields/VariableField.js

```diff
@@ -118,3 +118,12 @@ export default class VariableField extends Component<Props, State> {
     );
   }
 }
+
+export const VariableWithIcon = (value, iconPath, tooltip = '') => {
```

Comment:
* If you want to make a component, then the parameters should be `props`. Then you should use JSX to render it: `return <VariableWithIcon value={value} iconPath="res/types/scenevar.png" tooltip="scene variable" />`
* If you want a function, name it like a function: renderVariableWithIcon.

Here you've made a strange mix between those ;) It works, but convention is that if your function is named like a React component (PascalCase), then it should be used like a React component.

Reply:
Considering what you've done for objects (you've done a renderXXX function), this should then be named `renderVariableWithIcon` for consistency.